src/components/views/DownloadModModal.vue

```diff
@@ -167,112 +136,60 @@ interface DownloadProgress {
-                modName: '',
-                assignId: currentAssignId,
-                failed: false,
-            };
-            this.$store.commit('download/setDownloadObject', progressObject);
-            this.$store.commit('download/pushDownloadObjectToAllVersions', {
-                assignId: currentAssignId,
-                downloadObject: progressObject
-            });
             this.downloadingMod = true;
             ThunderstoreDownloaderProvider.instance.downloadLatestOfAll(modsWithUpdates, this.ignoreCache, (progress: number, modName: string, status: number, err: R2Error | null) => {
-                const assignIndex = this.$store.state.download.allVersions.findIndex(([number, val]: [number, DownloadProgress]) => number === currentAssignId);
-                if (status === StatusEnum.FAILURE) {
-                    if (err !== null) {
+                try {
+                    if (status === StatusEnum.FAILURE) {
                         this.downloadingMod = false;
-                        const existing = this.$store.state.download.allVersions[assignIndex];
-                        existing[1].failed = true;
-                        this.$store.commit('download/updateDownloadObject', {
-                            assignId: assignIndex,
-                            downloadVersion: [currentAssignId, existing[1]]
-                        });
-                        DownloadModModal.addSolutionsToError(err);
-                        this.$store.commit('error/handleError', err);
-                        return;
-                    }
-                } else if (status === StatusEnum.PENDING) {
-                    const obj = {
-                        progress: progress,
-                        modName: modName,
-                        initialMods: modsWithUpdates.map(value => `${value.getMod().getName()} (${value.getVersion().getVersionNumber().toString()})`),
-                        assignId: currentAssignId,
-                        failed: false,
-                    }
-                    if (this.$store.state.download.downloadObject!.assignId === currentAssignId) {
-                        this.$store.commit('download/setDownloadObject', obj);
+                        this.$store.commit('download/updateDownload', {assignId, failed: true});
+                        if (err !== null) {
+                            DownloadModModal.addSolutionsToError(err);
+                            throw err;
+                        }
+                    } else if (status === StatusEnum.PENDING) {
+                        this.$store.commit('download/updateDownload', {assignId, progress, modName});
                     }
-                    this.$store.commit('download/updateDownloadObject', {
-                        assignId: assignIndex,
-                        downloadVersion: [currentAssignId, obj]
-                    });
+                } catch (e) {
+                    this.$store.commit('error/handleError', e);
```

Comment:
Now that the "if is R2Error" block is removed per my earlier request, `e` is `unknown` type and shouldn't be passed to `handleError` as is (there's no complaints about it due to the poor type support with this Vuex version). Instead, use `R2Error.fromThrownValue(e)`